AuraFrontend/Program.cs

```diff
@@ -53,7 +54,23 @@ static void Main(string[] args)
 
 			CheckForHpd();
 
-			var recompileRequired = new UpdateSource(AuraDir, GitClonePath).Update();
+			if (!File.Exists(GitUrlPath))
+			{
+				PrintError("URL file 'git_url.txt' not found.");
+				Exit(true);
+			}
+
+			bool recompileRequired;
+			try
+			{
+				var gitClonePath = File.ReadAllText(GitUrlPath).Trim();
+				recompileRequired = new UpdateSource(AuraDir, gitClonePath).Update();
+			}
+			catch (LibGit2Sharp.LibGit2SharpException ex)
+			{
+				PrintError("Failed to download or update Aura source, startup aborted (Error: {0})", ex.Message);
+				Exit(true);
+			}
 
 			recompileRequired = true; // Hue for the hack.
```

Comment:
Not a clue. I assume originally you only wanted to recompile if necessary, but then you wanted to always do it, and that's actually fine because the way you compile only recompiles if necessary, I don't know.